### PR TITLE
DM-34479: replace descr argument with kwargs in BaseScript.amain and make_from_cmd_line class methods

### PR DESCRIPTION
**Pull request by r-owen**

### CODE REVIEW

File: python/lsst/ts/salobj/base_script.py [hunk 1]
```diff
@@ -214,49 +208,72 @@ def make_from_cmd_line(
...
+            if the command specifies ``--schema``.
 
         Notes
         -----
         The final return code will be:
 
         * 0 if final state is `lsst.ts.idl.enums.Script.ScriptState.DONE`
-          or `lsst.ts.idl.enums.Script.ScriptState.STOPPED`
+          or `lsst.ts.idl.enums.Script.ScriptState.STOPPED`.
         * 1 if final state is `lsst.ts.idl.enums.Script.ScriptState.FAILED`
```
Comment (wvreeven) [inline python/lsst/ts/salobj/base_script.py:238]: I see that you added a terminating `.` to line 237 so maybe add one here too?

File: python/lsst/ts/salobj/base_script.py [hunk 2]
```diff
@@ -214,49 +208,72 @@ def make_from_cmd_line(
...
-        * 2 otherwise (which should never happen)
+        * 2 if the final script state is other than DONE, STOPPED, or FAILED,
+            or if the final script state is DONE but script cleanup raises.
+            This should never happen
```
Comment (wvreeven) [inline python/lsst/ts/salobj/base_script.py:241]: I see that you added a terminating `.` to line 237 so maybe add one here too?

File: python/lsst/ts/salobj/base_script.py [hunk 3]
```diff
@@ -214,49 +208,72 @@ def make_from_cmd_line(
...
+            Keyword arguments for the script constructor.
+            Must not include ``index``.
+            Ignored (other than testing for ``index``)
+            if the command specifies ``--schema``.
 
         Notes
         -----
         The final return code will be:
 
         * 0 if final state is `lsst.ts.idl.enums.Script.ScriptState.DONE`
```
Comment (wvreeven) [inline python/lsst/ts/salobj/base_script.py:236]: These return codes look like ideal candidates to extract into an enum. Maybe for a future ticket?

**Review by wvreeven — APPROVED** [review]
Looks good to me. I left a few comments for improvements and leave it to you to process or not.

File: python/lsst/ts/salobj/base_script.py (hunk 3 above)
Comment (r-owen) [reply python/lsst/ts/salobj/base_script.py:236]: I am actually leaning towards removing all these special return codes and simply returning 1. The ScriptQueue is not using them and I can't think of a use for them. The script reports its final state using SAL.

Comment (r-owen) [reply python/lsst/ts/salobj/base_script.py:237]: I decided to simplify exit codes to 0/1 after looking at ts_scriptqueue and finding no usage of the exit code. All unit tests pass in ts_scriptqueue.

**Pull request merged**